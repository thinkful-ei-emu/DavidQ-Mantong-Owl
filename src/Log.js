import React from 'react';
import STORE from './STORE';

export default function Log(props){
  let when = new Date().toLocaleTimeString();
  return(<span>{props.name} {when}{props.action}{props.icon}</span>);
}