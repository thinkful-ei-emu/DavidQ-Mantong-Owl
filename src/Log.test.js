import ReactDOM from 'react-dom';
import React from 'react';
import renderer from 'react-test-renderer';

import Log from './Log';
describe('<Log />', () => {
  //smoke testing
  it ('render without crashing', ()=> {
      const div =document.createElement('div');
      ReactDOM.render(<Log name="David" action="has left" />,div);

      ReactDOM.unmountComponentAtNode(div);
  });
  //snapshot testing
  it ('renders this UI as expected', ()=>{
      const tree = renderer.create(<Log name="David" action="has left" />).toJSON();
      expect(tree).toMatchSnapshot();     
  });
});