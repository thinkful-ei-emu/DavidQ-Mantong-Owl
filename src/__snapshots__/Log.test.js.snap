exports[`<Log /> renders this UI as expected 1`] = `
<span>
  David
  Tue, 25 Jun 2019 20:44:31 GMT
  has left
</span>
`;
